contracts/libraries/OracleLibrary.sol

```diff
@@ -76,4 +76,34 @@ library OracleLibrary {
 
         return uint32(block.timestamp) - observationTimestamp;
     }
+
+    /// @notice Given a pool, it returns the tick value as of the start of the current block
+    /// @param pool Address of Uniswap V3 pool
+    /// @return The tick that the pool was in at the start of the current block
+    function getBlockStartingTick(address pool) internal view returns (int24) {
+        (, int24 tick, uint16 observationIndex, uint16 observationCardinality, , , ) = IUniswapV3Pool(pool).slot0();
```

Comment:
let's add an `observationCardinality > 0` check to ensure that the pool is initialized